fuse_models/include/fuse_models/unicycle_2d.h

```diff
@@ -133,13 +153,30 @@ class Unicycle2D : public fuse_core::AsyncMotionModel
    *                             \p beginning_stamp is guaranteed to be less than \p ending_stamp.
    * @param[in]  ending_stamp    The ending timestamp of the motion model constraints to be generated.
    *                             \p ending_stamp is guaranteed to be greater than \p beginning_stamp.
+   * @param[in]  state_history   The state history object to be updated
+   * @param[in]  device_id       The UUID of the device to be published
+   * @param[in]  name            The unique name for this motion model instance
+   * @param[in]  process_noise_covariance Process noise covariance matrix
+   * @param[in]  scale_process_noise  Whether to scale the process noise covariance pose by the norm
+                                      of the current state twist.
+   * @param[in]  velocity_norm_min  The minimum velocity/twist norm allowed when scaling the
+                                    process noise covariance
+   * @param[in]  disable_checks  Whether to disable the validation checks for the current and predicted state,
+                                 including the process noise covariance after it is scaled and multiplied by dt.
    * @param[out] constraints     One or more motion model constraints between the requested timestamps.
    * @param[out] variables       One or more variables at both the \p beginning_stamp and \p ending_stamp. The
    *                             variables should include initial values for the optimizer.
    */
-  void generateMotionModel(
+  static void generateMotionModel(
```

Comment:
I wonder if this breaks existing code because it might break the API for children classes. :thinking:

Reply:
Why did you have to make this `static`?

Reply:
I made this function static because I am honestly not that familiar with gtest and setting the private variables `process_noise_covariance_`, `state_history_`, etc. seemed complicated otherwise. Also `Unicycle2D::onInit()` gets some ROS parameters that again seemed more complicated to handle. Defining that function as static which I can then directly use seemed easier for me. 
However, I just thought about it again and with another way I came around this issue and there is no need to set this static and make many changes in the class itself (and also many comments from you are no more relevant then). If you feel like having a static function is still better, I can just revert my changes again. I look forward to your feedback on it.